### PR TITLE
adding glicko algortihm and unit tests for elo calculator

### PR DESCRIPTION
**Pull request by pmhansen8**
Changed the former elo algorithm to the Glicko algorithm which provides a more accurate ranking for players. Furthermore, to accommodate for the new algorithm I changed the former unit tests.

### CODE REVIEW

File: src/main/java/com/example/botfightwebserver/elo/EloCalculator.java [hunk 1]
```diff
@@ -6,52 +6,233 @@
...
+        if (player1.getElo() == null || player2.getElo() == null) {
+            throw new IllegalArgumentException("Player Elo cannot be null.");
         }
-        return K_FACTOR_DEFAULT;
+
+
+        Glicko glicko = new Glicko();
+
+        // Create player ratings
+        Glicko.Rating player1Rating = new Glicko.Rating(player1.getElo(), 350, 0.06);
```
Comment (Benler123) [inline src/main/java/com/example/botfightwebserver/elo/EloCalculator.java]: Can this be static so you don't have to initalize the glicko object?

File: src/main/java/com/example/botfightwebserver/elo/EloCalculator.java [hunk 2]
```diff
@@ -6,52 +6,233 @@
...
+        } else if (matchStatus == MATCH_STATUS.PLAYER_ONE_WIN) {
+            glickoMatchStatus = "player1";
+        } else if (matchStatus == MATCH_STATUS.PLAYER_TWO_WIN) {
+            glickoMatchStatus = "player2";
+        } else {
+            throw new IllegalArgumentException("Unknown match status: " + matchStatus);
+        }
+
+
+        Glicko.MatchResult result = glicko.rate1vs1(player1Rating, player2Rating, glickoMatchStatus);
```
Comment (Benler123) [inline src/main/java/com/example/botfightwebserver/elo/EloCalculator.java]: What does setting this here do?

Comment (pmhansen8) [reply src/main/java/com/example/botfightwebserver/elo/EloCalculator.java]: it stores the results of the phi sigma and elo change in an object making it easier to grab

File: src/main/java/com/example/botfightwebserver/elo/EloCalculator.java (hunk 1 above)
Comment (pmhansen8) [reply src/main/java/com/example/botfightwebserver/elo/EloCalculator.java]: Done!

**Pull request merged**